### PR TITLE
feature/#957 attempt to clean dn inheritance

### LINKED ISSUE #957: Clean DN inheritance
The inheritance mechanism for all data nodes would benefit from some cleaning.

Suggestions:

- The `_AbstractTabularDataNode` is not a data node but a mixin. The name is misleading. It is more like a mixin.
- Same for `_AbstractFileDataNode`.
- However, the `_abstract_sql` is a data node sub-class. Why this inconsistency?
- The `FilterDataNode` is not a data node sub-class nor a mixin. It is just a delegated class. I guess it should be renamed to `DataNodeFilter`.
- The exposed type code is duplicated in many data node implementations. I believe we can factorize the code in the `_AbstractTabularDataNode` class since the exposed type is for tabular data.

### PR DESCRIPTION
**Pull request by toan-quach**
#957

### CODE REVIEW

File: taipy/core/data/_abstract_file.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import shutil
 
 
-class _AbstractFileDataNode(object):
+class _AbstractFileDataNodeMixin(object):
```
Comment (jrobinAV) [inline taipy/core/data/_abstract_file.py]: At the end, it is not abstract and it is not a data node.

```suggestion
class _FileMixin(object):
```

File: taipy/core/data/_abstract_file.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
...
     """Abstract base class for data node implementations (CSVDataNode, ParquetDataNode, ExcelDataNode,
     PickleDataNode and JSONDataNode) that are file based."""
```
Comment (jrobinAV) [inline taipy/core/data/_abstract_file.py]:
```suggestion
    """Mixin class designed to handle file-based data nodes (CSVDataNode, ParquetDataNode, ExcelDataNode, PickleDataNode, JSONDataNode, etc.)."""
```

File: taipy/core/data/_abstract_tabular.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 from ..exceptions.exceptions import InvalidExposedType
 
 
-class _AbstractTabularDataNode(object):
+class _AbstractTabularDataNodeMixin(object):
```
Comment (jrobinAV) [inline taipy/core/data/_abstract_tabular.py]:
```suggestion
class _TabularMixin(object):
```

File: taipy/core/data/_abstract_tabular.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
...
     """Abstract base class for tabular data node implementations (CSVDataNode, ParquetDataNode, ExcelDataNode,
     SQLTableDataNode and SQLDataNode) that are tabular representable."""
```
Comment (jrobinAV) [inline taipy/core/data/_abstract_tabular.py]: We can change the docstring to make it up-to-date regarding the new name, no?

File: taipy/core/data/_abstract_tabular.py [hunk 5]
```diff
@@ -53,9 +53,20 @@ def _convert_data_to_dataframe(self, exposed_type: Any, data: Any) -> Union[pd.D
...
+    @classmethod
+    def _get_valid_exposed_type(cls, properties: Dict):
+        if (
+            cls._EXPOSED_TYPE_PROPERTY not in properties.keys()
+            or properties[cls._EXPOSED_TYPE_PROPERTY] == cls._EXPOSED_TYPE_MODIN
+        ):
+            # Default exposed type is pandas
+            # Deprecated modin exposed type in favor of pandas since 3.1.0
+            return cls._EXPOSED_TYPE_PANDAS
+        return properties[cls._EXPOSED_TYPE_PROPERTY]
```
Comment (jrobinAV) [inline taipy/core/data/_abstract_tabular.py:65]: Nice!

File: taipy/core/data/_abstract_file.py (hunk 1 above)
Comment (FabienLelaquais) [reply taipy/core/data/_abstract_file.py]: On the other hand... _FileMixin is really really vague...
But local to core/data... I'm good with that

**Review by FabienLelaquais — DISMISSED** [review]
To me, after the renaming as proposed by JR, that's good.
It IS a mixin indeed, and not an abstract class...

File: taipy/core/data/_abstract_tabular.py (hunk 5 above)
Comment (toan-quach) [reply taipy/core/data/_abstract_tabular.py:65]: about time 🤣

File: taipy/core/data/_abstract_file.py (hunk 1 above)
Comment (toan-quach) [reply taipy/core/data/_abstract_file.py]: how about `_FileDataNodeMixin`?

File: taipy/core/data/_abstract_tabular.py (hunk 3 above)
Comment (toan-quach) [reply taipy/core/data/_abstract_tabular.py]: I'd lean on `_TabularDataNodeMixin` since it's exclusively for dn use only. What do you think?

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**